src/System.Text.Json/tests/Serialization/PropertyVisibilityTests.cs

```diff
@@ -122,6 +124,128 @@ public static void JsonIgnoreAttribute()
             Assert.Equal(1, obj.MyDictionaryWithIgnore["Key"]);
         }
 
+        [Fact]
+        public static void JsonIgnoreAttribute_UnsupportedCollection()
+        {
+            string json =
+                    @"{
+                        ""MyConcurrentDict"":{
+                            ""key"":""value""
+                        },
+                        ""MyIDict"":{
+                            ""key"":""value""
+                        },
+                        ""MyDict"":{
+                            ""key"":""value""
+                        }
+                    }";
+            string wrapperJson =
+                    @"{
+                        ""MyClass"":{
+                            ""MyConcurrentDict"":{
+                                ""key"":""value""
+                            },
+                            ""MyIDict"":{
+                                ""key"":""value""
+                            },
+                            ""MyDict"":{
+                                ""key"":""value""
+                            }
+                        }
+                    }";
+
+            // Unsupported collections will throw by default.
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<ClassWithUnsupportedDictionary>(json));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(new ClassWithUnsupportedDictionary()));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Deserialize<WrapperForClassWithUnsupportedDictionary>(wrapperJson));
+            Assert.Throws<NotSupportedException>(() => JsonSerializer.Serialize(new WrapperForClassWithUnsupportedDictionary()));
+
+            // When ignored, we can serialize and deserialize without exceptions.
+            ClassWithIgnoredUnsupportedDictionary obj = JsonSerializer.Deserialize<ClassWithIgnoredUnsupportedDictionary>(json);
+            Assert.Null(obj.MyDict);
+            Assert.Equal("{}", JsonSerializer.Serialize(new ClassWithIgnoredUnsupportedDictionary()));
```

Comment:
In order to prevent previously cached metadata, these should use a new JsonSerializerOptions instance (for both serialize and deserialize - a separate instance for each)

Reply:
https://github.com/dotnet/corefx/pull/40414 addresses this and the changes from the PR have been ported to https://github.com/dotnet/corefx/pull/40413/commits/1c81d39f34bad62bfe27379c93d2c19b53bd3301.